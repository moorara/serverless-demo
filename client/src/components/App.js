import './App.css'
import logo from './logo.png'

import React, { Component } from 'react'

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Serverless App</h1>
        </header>
        <p className="App-intro">
          Serverless Client Application!
        </p>
      </div>
    )
  }
}

export default App
