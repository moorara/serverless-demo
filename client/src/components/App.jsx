import 'bulma/css/bulma.css'
import 'font-awesome/css/font-awesome.css'

import React, { Component } from 'react'
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'

import Header from './common/Header'
import Home from './home/Home'
import About from './about/About'
import NotFound from './common/NotFound'

class App extends Component {
  constructor (props) {
    super(props)
    this.state = {
      isLoading: false
    }
    this.setIsLoading = this.setIsLoading.bind(this)
  }

  setIsLoading (isLoading) {
    this.setState({ isLoading })
  }

  render() {
    return (
      <Router>
        <div>
          <Header isLoading={this.state.isLoading} />
          <Switch>
            <Route exact path="/" render={_ => <Home setIsLoading={this.setIsLoading} />} />
            <Route exact path="/about" component={About} />
            <Route component={NotFound} />
          </Switch>
        </div>
      </Router>
    )
  }
}

export default App
