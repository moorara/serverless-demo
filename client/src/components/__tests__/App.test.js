import React from 'react'
import renderer from 'react-test-renderer'

import App from '../App'

describe('App', () => {
  it('renders react component', () => {
    const component = renderer.create(<App />)
    const tree = component.toJSON()
    const instance = component.root

    expect(tree).toMatchSnapshot()
  })
})
