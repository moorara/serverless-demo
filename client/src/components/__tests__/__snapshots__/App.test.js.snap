// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders react component 1`] = `
<div>
  <section
    className="hero is-primary is-medium is-bold"
  >
    <div
      className="hero-head"
    >
      <div
        className="container"
      >
        <div
          className="columns"
        >
          <div
            className="column is-2 is-offset-5"
          >
            <div
              className="dots"
            >
              <span>
                ••
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <div
          className="columns"
        >
          <div
            className="column is-1"
          >
            <img
              alt="logo"
              className="app-logo"
              src="logo.png"
            />
          </div>
          <div
            className="column"
          >
            <h1
              className="title"
            >
              Serverless Application
            </h1>
            <h2
              className="subtitle"
            >
              A React client for consuming serverless API!
            </h2>
          </div>
        </div>
      </div>
    </div>
    <div
      className="hero-foot"
    >
      <nav
        className="tabs is-boxed"
      >
        <div
          className="container"
        >
          <ul>
            <li
              className="is-active"
            >
              <a
                aria-current="page"
                className="is-active"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                Home
              </a>
            </li>
            <li
              className=""
            >
              <a
                aria-current={null}
                href="/about"
                onClick={[Function]}
              >
                About
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </section>
  <section
    className="hero is-info is-medium"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Message Panel
        </h1>
        <div
          className="content"
        >
          <p>
            
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;
